### PR TITLE
refactor(steam,agg): use `insert` or `update` according to previous state instead of `upsert`

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

With #5598, we now have a single result state table for all agg calls that produce single value state (previously called _managed value state_), hence it's much simpler to decide whether to use `insert` or `update` when writing state table than before. So in this PR, the `upsert` method added in #5565 is removed and `insert`/`update` is used instead.

It's also made possible to enable sanity check after this PR, the changes to enable sanity check are submitted separately in #5709.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

https://github.com/risingwavelabs/risingwave/issues/3885#issuecomment-1269824531

### CODE REVIEW

File: src/stream/src/executor/aggregation/agg_state.rs [hunk 1]
```diff
@@ -176,17 +193,17 @@ impl<S: StateStore> AggState<S> {
...
-                .chain(curr_outputs.iter())
-                .cloned()
-                .collect(),
-        );
-        self.prev_outputs = Some(curr_outputs);
+        let result_row = self
+            .group_key
+            .as_ref()
+            .unwrap_or_else(Row::empty)
+            .concat(curr_outputs.clone().into_iter());
```
Comment (BugenZhao) [inline src/stream/src/executor/aggregation/agg_state.rs:200]:
```suggestion
            .concat(curr_outputs.iter().cloned());
```

**Review by BugenZhao — APPROVED** [review]
LGTM!

File: src/stream/src/executor/aggregation/agg_state.rs (hunk 1 above)
Comment (BugenZhao) [reply src/stream/src/executor/aggregation/agg_state.rs:200]: Same for other occurrences. :)

Comment (richardchien) [reply src/stream/src/executor/aggregation/agg_state.rs]: No other occurrence (

Comment (BugenZhao) [reply src/stream/src/executor/aggregation/agg_state.rs]: Oops. BTW, the `into_iter()` can also be eliminated.

**Pull request merged**